test_pool/pe/pe016.c

```diff
@@ -43,34 +43,40 @@ payload()
   tmp_reg_data = (sve_reg_details *)buffer_ptr;
   tmp_reg_data = tmp_reg_data + index;
 
+ /* Read ID_AA64ZFR0_EL1 for SVE2 support */
+  data = val_pe_reg_read(ID_AA64ZFR0_EL1);
+  tmp_reg_data->data = data;
+
+  /* For Armv9, the ID_AA64ZFR0_EL1.SVEver, bits [3:0] value 0b0000 is not permitted */
+  /* ID_AA64ZFR0_EL1.SVEver > 0 indicates FEAT_SVE2 or grater is implemented*/
+  /* If PE implements SVE2, it's a pass. No need to check architecture family, as
+   * SVE2 is required from v9 onwards. */
```

Comment:
I would rewrite that comment.

ArmArm L.b, page A2-158:

> FEAT_SVE2 is OPTIONAL from Armv9.0.

It is completely legal to have Armv9.0 cpu without any SVE. In such case ID_AA64ZFR0_EL1 register should not be present.


On the other side, BSA 1.2 mandates SVE2 for Armv9 cpus:

> Implementation of SVE is optional. Implementation of SVE2 (FEAT_SVE2) is required for PEs that are based on the Armv9 architecture (see FEAT_SVE2 in [3]).